python/GafferUI/EditScopeUI.py

```diff
@@ -265,14 +269,20 @@ def __inputNode( self ) :
 
 		node = self.getPlug().node()
 		# We assume that our plug is on a node dedicated to holding settings for the
-		# UI, and that it has an `in` plug that is connected to the node in the graph
+		# UI, and if the node has an `in` plug, it is connected to the node in the graph
 		# that is being viewed. We start our node graph traversal at the viewed node
 		# (we can't start at _this_ node, as then we will visit our own input connection
 		# which may no longer be upstream of the viewed node).
-		if node["in"].getInput() is None :
-			return None
+		if node.getChild( "in" ) :
+			if node["in"].getInput() is None :
+				return None
+
+			inputNode = node["in"].getInput().node()
+		else :
+			# Our node doesn't have an `in` plug so fall back to using the focus node
+			# as the starting point for node graph traversal.
+			inputNode = GafferUI.ContextTracker.acquireForFocus( node ).targetNode()
 
-		inputNode = node["in"].getInput().node()
 		if not isinstance( inputNode, Gaffer.EditScope ) and isinstance( inputNode, Gaffer.SubGraph ) :
```

Comment:
D'oh! Thanks for catching that. Rather than moving the block, we want this behaviour for both cases, so I've revamped things so it is no longer dependent on `node["in"]` and instead works from the output of `inputNode`. 30ac0a92546257dea70adc8bcb25f8f563b3bde4